client/src/components/Ratings/ReviewList.jsx

```diff
@@ -2,59 +2,110 @@ import React, { useState, useContext, useEffect } from 'react';
+  });
 
+  // totalCT get from reviewMeta isn't accurate due to reported reviews removal from db
   const getReview = () => (
     axios({
       method: 'get',
       url: '/reviews',
       params: {
         product_id: productId,
-        count: 2,
-        page,
+        count: reviewMeta?.totalCT,
+        sort,
       },
     }));
 
-  // TODO: check to see if there's memory leakage on unmounted components. Unsure why I didn't need a loading varable here. Will have to investigate
+  const addHelpVote = (reviewId) => {
+    axios({
+      method: 'put',
+      url: `/reviews/${reviewId}/helpful`,
+    })
+      .catch((err) => Console.log(err));
+  };
+
+  const handleSort = (criteria) => {
+    setSort(criteria);
+  }
+
+  const reportReview = (index, reviewId) => {
+    axios({
+      method: 'put',
+      url: `/reviews/${reviewId}/report`,
+    })
+    .then( () => {
+      let tempData = reviewDetail;
+      tempData.allReview.splice(index, 1);
+      setReviewDetail({
+        prevCount:reviewDetail.prevCount,
+        allReview: tempData.allReview,
+        totalCT: tempData.totalCT,
+      });
+    })
+      .catch((err) => Console.log(err));
+  };
+
+  const fetchFeed = () => {
+    if (reviewDetail.prevCount < reviewDetail.allReview.length) {
+      setReviewDetail({
+        allReview: reviewDetail.allReview,
+        prevCount: reviewDetail.allReview.length,
+        totalCT: reviewDetail.totalCT,
+      });
+    }
+  };
+
+  // TODO: check to see if there's memory leakage on unmounted components.
   useEffect(() => {
     getReview()
       .then(({ data }) => {
-        const review = data.results;
-        setPrevCount(review.length);
-        setReviewFeed(reviewFeed.concat(review));
+        setReviewDetail({
+          allReview: data.results,
+          totalCT: data.results.length,
+          prevCount: Math.min(data.results.length, 2),
+        });
       })
-      .catch((err) => console.log(err));
-  }, [page]);
+      .catch((err) => Console.log(err));
+  }, [productId, sort]);
+
 
-  const fetchFeed = () => {
-    totalCT = reviewMeta?.totalCT;
-    if(prevCount < totalCT) {
-      setPage(page + 1);
-    }
-  };
 
   return (
     <ReviewContainer>
-    {reviewFeed.map(
-      (review) => (
-        <ReviewTile
-        key={review.review_id}
-        review={review}
-        />
+      <SortBar
+      totalCT={reviewDetail.totalCT}
+      handleSort={handleSort}/>
+      {reviewDetail.allReview.slice(0, reviewDetail.prevCount).map(
+        (review, index) => (
+          <ReviewTile
+            key={review.review_id.toString()}
+            addHelpVote={addHelpVote}
+            review={review}
+            index={index}
+            reportReview={reportReview}
+          />
         ),
       )}
       <ButtonBlock>
-        <Botton onClick={fetchFeed}> MORE REVIEWS</Botton>
+        {(reviewDetail.prevCount < reviewDetail.allReview.length)
+          ? (
+            <Botton onClick={fetchFeed}>
```

Comment:
Could this live outside of the `ReviewContainer`? It feels a little weird that if the initial 2 reviews are too long that this button gets cut off.

Reply:
Will take a look. That's a good point.